src/test/java/rx/plugins/RxJavaHooksTest.java

```diff
@@ -552,4 +603,458 @@ public void onScheduleNewThread() throws InterruptedException {
         onSchedule(Schedulers.newThread().createWorker());
     }
 
+    @Test
+    public void onError() {
+        try {
+            final List<Throwable> list = new ArrayList<Throwable>();
+            
+            RxJavaHooks.setOnError(new Action1<Throwable>() {
+                @Override
+                public void call(Throwable t) {
+                    list.add(t);
+                }
+            });
+            
+            RxJavaHooks.onError(new TestException("Forced failure"));
+            
+            assertEquals(1, list.size());
+            assertTestException(list, 0, "Forced failure");
+        } finally {
+            RxJavaHooks.reset();
+        }
+    }
+    
+    @Test
+    public void clear() throws Exception {
+        RxJavaHooks.reset();
+        try {
+            RxJavaHooks.clear();
+            for (Method m : RxJavaHooks.class.getMethods()) {
+                if (m.getName().startsWith("getOn")) {
+                    assertNull(m.toString(), m.invoke(null));
+                }
+            }
+
+        } finally {
+            RxJavaHooks.reset();
+        }
+
+        for (Method m : RxJavaHooks.class.getMethods()) {
+            if (m.getName().startsWith("getOn") && !m.getName().endsWith("Scheduler")) {
+                assertNotNull(m.toString(), m.invoke(null));
+            }
+        }
+
+    }
+
+    @Test
+    public void onErrorNoHandler() {
+        try {
+            final List<Throwable> list = new ArrayList<Throwable>();
+            
+            RxJavaHooks.setOnError(null);
+            
+            Thread.currentThread().setUncaughtExceptionHandler(new UncaughtExceptionHandler() {
+                
+                @Override
+                public void uncaughtException(Thread t, Throwable e) {
+                    list.add(e);
+                    
+                }
+            });
+            
+            RxJavaHooks.onError(new TestException("Forced failure"));
+
+            Thread.currentThread().setUncaughtExceptionHandler(null);
+
+            // this will be printed on the console and should not crash
```

Comment:
You could actually test this by setting `System.err` to something that records, but it would be a bit much.